lib/cocoapods/sandbox/path_list.rb

```diff
@@ -128,19 +128,21 @@ def relative_glob(patterns, options = {})
         else
           full_list = files
         end
-
-        list = Array(patterns).map do |pattern|
+        patterns_array = Array(patterns)
+        exact_matches = full_list & patterns_array
+        patterns_array -= exact_matches
+        all_patterns = patterns_array.map do |pattern|
           if directory?(pattern) && dir_pattern
             pattern += '/' unless pattern.end_with?('/')
             pattern += dir_pattern
           end
-          expanded_patterns = dir_glob_equivalent_patterns(pattern)
-          full_list.select do |path|
-            expanded_patterns.any? do |p|
-              File.fnmatch(p, path, File::FNM_CASEFOLD | File::FNM_PATHNAME)
-            end
-          end
+          dir_glob_equivalent_patterns(pattern)
         end.flatten
+        list = exact_matches + full_list.select do |path|
```

Comment:
why can probably skip the `full_list.select` if `all_patterns.empty?`

Reply:
Ow yeah, will do.